client/me/purchases/concierge-banner/index.jsx

```diff
@@ -3,39 +3,137 @@
+					</div>
+					<div className="concierge-banner__placeholder-button-container">
+						<div className="concierge-banner__placeholder-button is-placeholder" />
+					</div>
+				</div>
+			</Card>
+		);
+	}
+
+	getText() {
+		const { bannerType } = this.props;
+
+		let headerText, mainText, buttonText, buttonHref, illustrationUrl;
+
+		switch ( bannerType ) {
+			case 'upcomingAppointment':
+				headerText = this.props.translate( 'Your appointment is coming up!' );
+				mainText = this.props.translate(
+					'Get ready with your questions for your upcoming Quick Start session appointment.',
+					{
+						comment:
+							"Please extend the translation so that it's clear that these sessions are only available in English.",
+					}
+				);
+				buttonText = this.props.translate( 'Session dashboard' );
+				buttonHref = '/me/concierge';
+				illustrationUrl = conciergeImage;
+				break;
+
+			case 'availableIncludedSession':
+				headerText = this.props.translate( 'Looking for Expert Help?' );
+				mainText = this.props.translate(
+					'Get 30 minutes dedicated to the success of your site. Schedule your free 1-1 Quick Start Session with a Happiness Engineer!',
+					{
+						comment:
+							"Please extend the translation so that it's clear that these sessions are only available in English.",
+					}
+				);
+				buttonText = this.props.translate( 'Schedule Now' );
+				buttonHref = '/me/concierge';
+				illustrationUrl = conciergeImage;
+				break;
+
+			case 'availablePurchasedSession':
+				headerText = this.props.translate( 'Our experts are waiting to help you' );
+				mainText = this.props.translate(
+					'Schedule your 45-minute 1-1 Quick Start Session with a Happiness Engineer!',
+					{
+						comment:
+							"Please extend the translation so that it's clear that these sessions are only available in English.",
+					}
+				);
+				buttonText = this.props.translate( 'Schedule Now' );
+				buttonHref = '/me/concierge';
+				illustrationUrl = conciergeImage;
+				break;
+
+			case 'canPurchaseConcierge':
+				headerText = this.props.translate( 'Need an expert by your side?' );
+				mainText = this.props.translate(
+					'We offer one-on-one Quick Start sessions dedicated to your site’s success. Click the button to learn how we can help you during these 45 minute calls.',
+					{
+						comment:
+							"Please extend the translation so that it's clear that these sessions are only available in English.",
+					}
+				);
+				buttonText = this.props.translate( 'Learn More' );
+				buttonHref = '/checkout/offer-quickstart-session';
+				illustrationUrl = '/calypso/images/illustrations/illustration-start.svg';
+				break;
+		}
+
+		return { headerText, mainText, buttonText, buttonHref, illustrationUrl };
+	}
+
 	render() {
+		const { bannerType } = this.props;
+
+		if ( 'placeholder' === bannerType ) {
+			return this.placeholder();
+		}
+
+		const { headerText, mainText, buttonText, buttonHref, illustrationUrl } = this.getText();
```

Comment:
Nitpick: given the comprehensiveness here, `getText()` doesn't capture the meaning quite well. Would something like `getBannerContent()` be better?

It's totally a soft comment, so please feel free to keep it as it is.

Reply:
Done in [4d6a1d9](https://github.com/Automattic/wp-calypso/pull/34956/commits/4d6a1d996874a6dff8f16889b1bf560f27a168fe)